### PR TITLE
Database model for user access tokens

### PR DESCRIPTION
**Pull request by lfaraone**

### CODE REVIEW

File: grouper/models.py [hunk 1]
```diff
@@ -631,6 +635,55 @@ def my_open_audits(self):
         ).all()
 
 
+def _make_secret():
+    return os.urandom(20).encode("hex")
+
+
+class UserToken(Model):
```
Comment (lfaraone) [inline grouper/models.py]: All of this needs some test cases...

File: tests/test_users.py [hunk 2]
```diff
@@ -44,6 +44,28 @@ def test_basic_metadata(standard_graph, session, users, groups, permissions):  #
     assert len(md) == 1, "One metadata item"
 
 
+def test_usertokens(standard_graph, session, users, groups, permissions):  # noqa
```
Comment (lfaraone) [inline tests/test_users.py]: Should I be doing something different to ensure I don't mutate shared test state?

File: tests/test_users.py [hunk 3]
```diff
@@ -44,6 +44,28 @@ def test_basic_metadata(standard_graph, session, users, groups, permissions):  #
...
+    user = users["zorkian@a.co"]
```
Comment (lfaraone) [inline tests/test_users.py]: Should I be doing something different to ensure I don't mutate shared test state?

File: grouper/models.py [hunk 4]
```diff
@@ -1,3 +1,6 @@
+import hmac
+
+import os
```
Comment (herb) [inline grouper/models.py]: nit: alphabetical please

File: grouper/models.py [hunk 5]
```diff
@@ -631,6 +635,59 @@ def my_open_audits(self):
...
+
+    __tablename__ = "user_tokens"
+
+    id = Column(Integer, primary_key=True)
+
+    user_id = Column(Integer, ForeignKey("users.id"))
+    name = Column(String(length=16), nullable=False)
+
+    ctime = Column(DateTime, default=datetime.utcnow, nullable=False)
+    dtime = Column(DateTime, default=None, nullable=True)
```
Comment (herb) [inline grouper/models.py]: i'm not a huge personal fan of the naming convention that exists but can we make these two more consistent with it? so `ctime` should be `created_on` or `created_at` and 'dtime' should be `disabled_on` or `disabled_at`

File: tests/test_users.py (hunk 3 above)
Comment (herb) [inline tests/test_users.py]: no, you're good. each test gets a new database and associated fixtures.

File: grouper/models.py (hunk 4 above)
Comment (lfaraone) [inline grouper/models.py]: Fixed.

File: grouper/models.py (hunk 5 above)
Comment (lfaraone) [inline grouper/models.py]: Fixed.

**Pull request merged**